firmware/ventilator-controller-stm32/Core/Test/Pufferfish/Protocols/messages.txt

```diff
@@ -0,0 +1,181 @@
+      THEN("The output buffer is as expected")
+/*
+
+/* Parse function
+
+Scenario: Message: The parse method correctly updates internal message type and sets the message payload fields after decoding from the input buffer
+
+  GIVEN("A Message object constructed with StateSegment Taggedunion, default message types and a max size of 252 bytes")
+
+    WHEN("A body with an encoded payload and 1 byte header of value equal to 0 is parsed")
+      THEN("The parse method reports invalid type status")
+      THEN("After the parse method is called, The value assigned to the type member is equal to the input_buffer body's header")
+      THEN("The payload Taggedunion tag is set to Unknown")
+      THEN("The message payload fields are not set")
+
+    WHEN("An empty input buffer body is parsed")
+      THEN("The parse method reports invalid length status")
+      THEN("The payload Taggedunion tag is set to Unknown")
+      THEN("The message payload fields are not set")
+
+    WHEN("A body with an empty payload and 1 byte header of value equal to the message descriptor array size is parsed")
+      THEN("The parse method reports invalid type status")
+      THEN("After the parse method is called, The value assigned to the type member is equal to the input_buffer body's header")
+      THEN("The payload Taggedunion tag is set to Unknown")
+      THEN("The message payload fields are not set")
+
+    WHEN("A body with an empty payload and 1 byte header of value greater than the message descriptor array size is parsed")
+      THEN("The parse method reports invalid type status")
+      THEN("After the parse method is called, The value assigned to the type member is equal to the input_buffer body's header")
+      THEN("The payload Taggedunion tag is set to Unknown")
+      THEN("The message payload fields are not set")
+
+    WHEN("A body with a 1 byte header and payload of sensor_measurements message with invalid encoding is parsed")
+      THEN("The parse method reports invalid encoding status")
+      THEN("After the parse method is called, The value assigned to the type member is equal to the input_buffer body's header")
+      THEN("The payload Taggedunion tag is set to sensor_measurements")
+      THEN("The message payload fields are not set")
+
+    WHEN("The input buffer size is greater than the payload max size")
+      THEN("The status of the parse function returns invalid encoding")
+
+/* test cases for parse functions for each message type
+    WHEN("A body with 1 byte header and a paylod of {Message type} message type is parsed")
+      THEN("The parse method reports ok status")
+      THEN("After the parse method is called, The value assigned to the type member is equal to the input_buffer body's header")
+      THEN("The payload Taggedunion tag is set to {Message type}")
+      THEN("The message payload field values are as expected")
```

Comment:
The language is a bit ambiguous. I think rewording it will resolve this: "THEN the payload.values field has the expected {Message type} field values" or "THEN the payload field's values field has the expected {Message type} field values" or something similar.

Whatever you choose, let's use the same pattern when we refer to the payload.tag field (alternately, "the payload field's tag field" or something similar).

Reply:
It looks like this comment was marked as resolved without addressing either of the suggestions in this comment?

Reply:
I had updated everywhere else and forgot to update here. fixed now.